src/Native/System.Native/pal_io.cpp

```diff
@@ -202,3 +214,83 @@ int32_t ShmUnlink(const char* name)
 {
     return shm_unlink(name);
 }
+
+
+static
+void ConvertDirent(const dirent& entry, DirectoryEntry* outputEntry)
+{
+    // We use Marshal.PtrToStringAnsi on the managed side, which takes a pointer to
+    // the start of the unmanaged string. Give the caller back a pointer to the 
+    // location of the start of the string that exists in their own byte buffer.
+    outputEntry->nameStartAddress = &entry.d_name[0];
```

Comment:
Style choice, but can just use `entry.d_name` rather than `&entry.d_name[0]`.and let array -> pointer decay do its things.